pyfar/dsp/filter.py

```diff
@@ -679,3 +680,285 @@ def _shelve(signal, frequency, gain, order, shelve_type, sampling_rate, kind):
+
+
+def _exact_center_frequencies_fractional_octaves(
+        num_fractions, frequency_range):
+    """Calculate the center frequencies of arbitrary fractional octave bands.
+
+    Parameters
+    ----------
+    num_fractions : int
+        The number of fractions
+    frequency_range
+        The upper and lower frequency limits
+
+    Returns
+    -------
+    exact : array, float
+        An array containing the center frequencies of the respective fractional
+        octave bands
+
+    """
+    ref_freq = 1e3
+    Nmax = np.around(num_fractions*(np.log2(frequency_range[1]/ref_freq)))
+    Nmin = np.around(num_fractions*(np.log2(ref_freq/frequency_range[0])))
+
+    indices = np.arange(-Nmin, Nmax+1)
+    exact = ref_freq * 2**(indices / num_fractions)
+
+    return exact
+
+
+def _exact_center_frequencies_fractional_octaves_iec(nominal, num_fractions):
+    """Returns the exact center frequencies for fractional octave bands
+    according to the IEC 61260:1:2014 standard.
+    octave ratio
+    .. G = 10^{3/10}
+    center frequencies
+    .. f_m = f_r G^{x/b}
+    .. f_m = f_e G^{(2x+1)/(2b)}
+    where b is the number of octave fractions, f_r is the reference frequency
+    chosen as 1000Hz and x is the index of the frequency band.
+
+    Parameters
+    ----------
+    indices : array
+        The indices for which the center frequencies are calculated.
+    num_fractions : 1, 3
+        The number of octave fractions. 1 returns octave center frequencies,
+        3 returns third octave center frequencies.
+    Returns
+    -------
+    frequencies : ndarray, float
+        center frequencies of the fractional octave bands
+    """
+    indices = _frequency_indices(nominal, num_fractions)
+    reference_freq = 1e3
+    octave_ratio = 10**(3/10)
+
+    iseven = np.mod(num_fractions, 2) == 0
+    if ~iseven:
+        exponent = (indices/num_fractions)
+    else:
+        exponent = ((2*indices + 1) / num_fractions / 2)
+
+    center_freq = reference_freq * octave_ratio**exponent
+
+    return center_freq
+
+
+def _frequency_indices(frequencies, num_fractions):
+    """Return the indices for fractional octave filters.
+    Parameters
+    ----------
+    frequencies : array
+        The nominal frequencies for which the indices for exact center
+        frequency calculation are to be calculated.
+    num_fractions : 1, 3
+        Number of fractional bands
+    Returns
+    -------
+    indices : array
+        The indices for exact center frequency calculation.
+    """
+    reference_freq = 1e3
+    octave_ratio = 10**(3/10)
+
+    iseven = np.mod(num_fractions, 2) == 0
+    if ~iseven:
+        indices = np.around(
+            num_fractions * np.log(frequencies/reference_freq)
+            / np.log(octave_ratio))
+    else:
+        indices = np.around(
+            2.0*num_fractions *
+            np.log(frequencies/reference_freq) / np.log(octave_ratio) - 1)/2
+
+    return indices
+
+
+def fractional_octave_bands(
```

Comment:
How about naming this only `fractional_octaves` or `fractional_octaves_iec`? I'm suggesting this with having the perfect reconstructing fractional octave filter bank in mind that should well separated by name. Having the `band` in there will make the names very long...